Source/MQTTnet/Formatter/MqttBufferReader.cs

```diff
@@ -34,52 +36,53 @@ public byte[] ReadBinaryData()
             ValidateReceiveBuffer(length);
 
             var result = new byte[length];
-            Array.Copy(_buffer, Position + _offset, result, 0, length);
-            Position += length;
+            Array.Copy(_buffer, _position, result, 0, length);
+            _position += length;
 
             return result;
         }
 
         public byte ReadByte()
         {
             ValidateReceiveBuffer(1);
-
-            return _buffer[Position++];
+            return _buffer[_position++];
         }
 
         public uint ReadFourByteInteger()
         {
             ValidateReceiveBuffer(4);
 
 #if NETCOREAPP3_0_OR_GREATER
-            var value = System.Runtime.InteropServices.MemoryMarshal.Read<uint>(_buffer.AsSpan().Slice(Position));
-            value = System.Buffers.Binary.BinaryPrimitives.ReverseEndianness(value);
+            var value = MemoryMarshal.Read<uint>(_buffer.AsSpan(_position));
+            if (BitConverter.IsLittleEndian)
```

Comment:
I intentionally avoided added the endian check because the old code (below) also does not revert the values if required.
So, what will happen when the old code is running on a little-endian machine?

Reply:
The old code is running on a little-endian machine correctly, but not on a big-endian machine.

Reply:
The network byte order is the same as the big-endian machine byte order, so reverse will get the wrong value.

Reply:
Left shift high bits is correct in all cases.
We can use [BinaryPrimitives.ReadUInt32BigEndian`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/ReaderBigEndian.cs#L112) method if the Api is supported,
```c#
public static uint ReadUInt32BigEndian(ReadOnlySpan<byte> source)
{
    return BitConverter.IsLittleEndian ?
        ReverseEndianness(MemoryMarshal.Read<uint>(source)) :
        MemoryMarshal.Read<uint>(source);
}
```

Reply:
Then let's use the new API then. Then I will merge the PR.